### PR TITLE
Comply with LibreNMS 1.40 requirements

### PR DESCRIPTION
**Pull request by inetAnt**
As LibreNMS 1.40 comes with [some important changes](https://community.librenms.org/t/new-requirements-for-1-40/4057/2), the Cookbook needs a refreshment to integrate those.

- Install PHP mbstring extension
- Double MySQL max_connections
- Create new directories
- Create .env file

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -207,6 +210,33 @@
   not_if { ::File.exist? librenms_rrddir }
 end
 
+directory librenms_bootstrap_cachedir do
+  owner librenms_username
+  group librenms_group
+  mode '0755'
+  action :create
+  not_if { ::File.exist? librenms_bootstrap_cachedir }
```
Comment (jmauro) [inline recipes/default.rb]: This is exactly what the resource `directory` does

File: recipes/default.rb [hunk 2]
```diff
@@ -207,6 +210,33 @@
...
+end
+
+directory librenms_storagedir do
+  owner librenms_username
+  group librenms_group
+  mode '0755'
+  action :create
+  not_if { ::File.exist? librenms_storagedir }
```
Comment (jmauro) [inline recipes/default.rb]: Same as above

File: recipes/default.rb [hunk 3]
```diff
@@ -207,6 +210,33 @@
...
+end
+
+template librenms_env_file do
```
Comment (jmauro) [inline recipes/default.rb]: Why a template? This is just a key/value format

File: templates/default/librenms-mysqld.cnf.erb [hunk 4]
```diff
@@ -4,3 +4,4 @@ sql-mode=""
 lower_case_table_names=0
 bind-address = <%= @bind_address %>
 innodb_flush_log_at_trx_commit = 0
+max_connections=302
```
Comment (jmauro) [inline templates/default/librenms-mysqld.cnf.erb]: A static value? Why not another key/value file?

**Review by jmauro — CHANGES_REQUESTED** [review]

File: recipes/default.rb (hunk 1 above)
Comment (inetAnt) [reply recipes/default.rb]: Hopefully fixed

File: recipes/default.rb (hunk 2 above)
Comment (inetAnt) [reply recipes/default.rb]: Hopefully fixed

File: templates/default/librenms-mysqld.cnf.erb (hunk 4 above)
Comment (inetAnt) [reply templates/default/librenms-mysqld.cnf.erb]: Done

Comment (inetAnt) [reply templates/default/librenms-mysqld.cnf.erb]: (added as a variable in the already existing template)

**Review by nicolasgrieco — APPROVED** [review]

**Review by jmauro — APPROVED** [review]

**Pull request merged**